### PR TITLE
chore(ui): Clean up settings component for shared uses

### PR DESCRIPTION
**Pull request by agg23**
Separated out changes for settings from #34010 and #34058.

### CODE REVIEW

**Comment by pavelfeldman** [comment]
Does it change UI? If it does, please attach before / after screenshots!

File: packages/trace-viewer/src/ui/settingsView.tsx [hunk 1]
```diff
@@ -18,22 +18,32 @@ import * as React from 'react';
...
-          <input type='checkbox' checked={value} onChange={() => set(!value)}/>
-          {title}
-        </label>
-      </div>;
-    })}
-  </div>;
-};
+  return (
+    <div className='vbox settings-view'>
+      {settings.map(({ value, set, name, title }) => {
```
Comment (pavelfeldman) [inline packages/trace-viewer/src/ui/settingsView.tsx:32]: If setting becomes a class, destructuring will cause problems (illegal this in set invocation)

File: packages/trace-viewer/src/ui/settingsView.tsx [hunk 2]
```diff
@@ -18,22 +18,32 @@ import * as React from 'react';
...
+        const labelId = `setting-${name}`;
+
+        return (
+          <div key={name} className='setting' title={title}>
+            <input
+              type='checkbox'
+              id={labelId}
+              checked={value}
+              onChange={() => set(!value)}
```
Comment (pavelfeldman) [inline packages/trace-viewer/src/ui/settingsView.tsx:41]: That'll probably cause re-render - you know better though!

File: packages/trace-viewer/src/ui/settingsView.css [hunk 3]
```diff
@@ -15,25 +15,27 @@
...
-.settings-view .setting:last-of-type label {
-  margin-bottom: 2px;
+  cursor: pointer;
 }
 
 .settings-view .setting input {
   margin-right: 5px;
-}
+  flex-shrink: 0;
+}
```
Comment (pavelfeldman) [inline packages/trace-viewer/src/ui/settingsView.css:41]: we like new lines in the end of file :P

**Review by pavelfeldman — APPROVED** [review]

File: packages/trace-viewer/src/ui/settingsView.tsx (hunk 2 above)
Comment (agg23) [reply packages/trace-viewer/src/ui/settingsView.tsx:41]: It does. There's a lot of rerender perf issues present right now. That will be a future PR if we decide to fix it.

**Pull request merged**